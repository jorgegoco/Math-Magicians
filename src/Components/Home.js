import React from 'react';

const Home = () => (
  <section className="homeSection">
    <h1>Welcome to our page!</h1>
    <p>
      &#34;Math magicians&#34; is a website for all fans of mathematics.
      It is a Single Page App &#40;SPA&#41; that allows users to:
    </p>
    <ul>
      <li>Make simple calculations.</li>
      <li>Read a random math-related quote.</li>
    </ul>
  </section>
);
export default Home;
