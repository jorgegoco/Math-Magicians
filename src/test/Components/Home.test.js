import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Home from '../../Components/Home';

describe('Home component test', () => {
  test('snapshot renders', () => {
    const home = renderer
      .create(<Home />)
      .toJSON();
    expect(home).toMatchSnapshot();
  });

  test('matches element', () => {
    render(<Home />);
    expect(screen.getByText(/Welcome to our page!/)).toBeInTheDocument();
  });
});
