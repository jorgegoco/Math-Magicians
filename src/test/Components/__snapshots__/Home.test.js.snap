// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component test snapshot renders 1`] = `
<section
  className="homeSection"
>
  <h1>
    Welcome to our page!
  </h1>
  <p>
    "Math magicians" is a website for all fans of mathematics. It is a Single Page App (SPA) that allows users to:
  </p>
  <ul>
    <li>
      Make simple calculations.
    </li>
    <li>
      Read a random math-related quote.
    </li>
  </ul>
</section>
`;
